### PR TITLE
[vcpkg baseline][openxr-loader] Fix config error

### PR DESCRIPTION
**Pull request by JonLiu1993**
Fix `qtquick3d` failed `Undefined _BrotliEncoderCompressStream`, https://dev.azure.com/vcpkg/public/_build/results?buildId=110038&view=results

Add function `find_dependency(jsoncpp CONFIG)` to port openxr-loader

### Checklist

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [ ] ~SHA512s are updated for each updated download.~
- [ ] ~The "supports" clause reflects platforms that may be fixed by this new version.~
- [ ] ~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~
- [ ] ~Any patches that are no longer applied are deleted from the port's directory.~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### Test

The port installation tests pass with the following triplets:

* x64-linux

### CODE REVIEW

File: ports/openxr-loader/fix-config-error.patch [hunk 1]
```diff
@@ -0,0 +1,12 @@
+diff --git a/src/loader/OpenXRConfig.cmake.in b/src/loader/OpenXRConfig.cmake.in
+index 81b12e7..4c24771 100644
+--- a/src/loader/OpenXRConfig.cmake.in
++++ b/src/loader/OpenXRConfig.cmake.in
+@@ -6,6 +6,7 @@
+ 
+ include(CMakeFindDependencyMacro)
+ find_dependency(Threads)
++find_dependency(jsoncpp CONFIG)
```
Comment (BillyONeal) [inline ports/openxr-loader/fix-config-error.patch:9]: Agreed; upstream says `NO_MODULE`:

https://github.com/KhronosGroup/OpenXR-SDK/blob/781f2eab3698d653c804ecbd11e0aed47eaad1c6/src/cmake/FindJsonCpp.cmake#L68

but this is documented to be the same as `CONFIG`: https://cmake.org/cmake/help/latest/command/find_package.html#full-signature

**Review by BillyONeal — APPROVED** [review]

**Comment by dg0yt** [comment]
> Fix `qtquick3d` failed `Undefined _BrotliEncoderCompressStream`, https://dev.azure.com/vcpkg/public/_build/results?buildId=110038&view=results

Unrelated to brotli.

qtquick3d still has an installation order problem. The recent openxr-loader update was pulled out of #42080, and I didn't notice the jsoncpp issue because there I disabled qtquick3d's use of openxr. qtquick3d is meant to get a feature to control openxr usage.

**Comment by dg0yt** [comment]
Cherry-picked into #42588.

**Comment by dg0yt** [comment]
Don't merge this.
Use https://github.com/microsoft/vcpkg/pull/42588 instead.

**Pull request closed without merging**